calico/election.py

```diff
@@ -93,14 +94,21 @@ def _run(self):
                     # Something failed, and wants us just to go back to the
                     # beginning.
                     pass
-
-                # Sleep a little before we go back and read again.
-                eventlet.sleep(self._interval)
-
-        except:
+                # In case we're repeatedly failing, sleep a little before we
+                # retry.
+                retry_time = 1 + (5 * random.random())
+                _log.info("Retrying leader election in %.1f seconds",
+                          retry_time)
+                eventlet.sleep(retry_time)
+        except ElectorStopped:
+            _log.info("Elector told to stop.")
+            raise
+        except BaseException as e:
             # Election greenlet failed. Log but reraise.
-            _log.exception("Election greenlet exiting")
+            _log.exception("Election greenlet exiting: %r", e)
```

Comment:
This change is unneeded: `log.exception` will dump out the entire exception stacktrace.